packages/cactus-plugin-ledger-connector-corda/src/main/typescript/plugin-ledger-connector-corda.ts

```diff
@@ -243,4 +324,210 @@ export class PluginLedgerConnectorCorda
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized,
+    });
+    return { headers, agent: httpsAgent };
+  }
+
+  private validateHoldingIDShortHash(inputString: string): boolean {
+    /* only hexadecimal characters and exactly 12 characters long as per the corda docs
+       https://docs.r3.com/en/platform/corda/5.2/developing-applications/cordapp-template/utxo-ledger-example-cordapp/running-the-chat-cordapp.html#using-swagger
+    */
+    const pattern = /^[0-9A-Fa-f]{12}$/;
+    return pattern.test(inputString);
+  }
+
+  public async startFlow(req: StartFlowV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      const holdingIDShortHash = req.holdingIDShortHash;
+      if (typeof holdingIDShortHash === "undefined") {
+        throw new Error("holdingIDShortHash is undefined");
+      }
+      if (!this.validateHoldingIDShortHash(holdingIDShortHash)) {
+        throw new Error(`Invalid holdingIDShortHash: ${holdingIDShortHash}`);
+      }
+      const cordaReq = {
+        clientRequestId: req.clientRequestId,
+        flowClassName: req.flowClassName,
+        requestBody: req.requestBody,
+      };
+      const cordaReqBuff = Buffer.from(JSON.stringify(cordaReq));
+      const startFlowUrl = urlcat(this.baseUrl, "flow/:holdingIDShortHash", {
+        holdingIDShortHash,
+      });
+      const response = await fetch(startFlowUrl, {
+        method: `POST`,
+        headers,
+        body: cordaReqBuff,
+        agent,
+      });
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const responseData = await this.pollEndpointUntilCompleted(
+        req.holdingIDShortHash!,
+        req.clientRequestId,
+      );
+      return responseData;
+    } catch (error) {
+      console.error("Error starting flow:", error);
+      throw error;
+    }
+  }
+
+  public async listCPI(req: ListCpiV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      const cpiUrl = urlcat(this.baseUrl, "cpi");
+      const response = await fetch(cpiUrl, {
+        method: `GET`,
+        headers: headers,
+        agent,
+      });
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const responseData = await response.json();
+      return responseData;
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    }
+  }
+
+  public async getFlow(req: GetFlowCidV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      if (typeof req.holdingIDShortHash === "undefined") {
+        throw new Error("holdingIDShortHash is undefined");
```

Comment:
@adrianbatuto Please throw a `BadRequestError` here fro the library mentioned in my other comment.